searx/network/__init__.py

```diff
@@ -1,266 +1,483 @@
+    * raise_for_httperror:
+        when True, this function calls searx.network.raise_for_httperror.raise_for_httperror.
+
+    Some parameters from httpx.Client ( https://www.python-httpx.org/api/#client) are available:
+
+    * max_redirects:
+        Set to None to use the value from the Network configuration.
+        The maximum number of redirect responses that should be followed.
+    * verify:
+        Set to None to use the value from the Network configuration.
+    * limits:
+        it has to be defined in the Network configuration (in settings.yml)
+    * default_encoding:
+        this parameter is not available and is always "utf-8".
+
+    This function requires a NetworkContext provided by either provide_networkcontext or networkcontext_for_thread.
+
+    The implementation uses one or more httpx.Client
+    """
+    # pylint: disable=too-many-arguments
+    network_context: Optional[NetworkContext] = getattr(_THREADLOCAL, _NETWORK_CONTEXT_KEY, None)
+    if network_context is None:
+        raise NetworkContextNotFound()
+    http_client = network_context._get_http_client()  # pylint: disable=protected-access
+    return http_client.request(
+        method,
+        url,
+        params=params,
+        content=content,
+        data=data,
+        files=files,
+        json=json,
+        headers=headers,
+        cookies=cookies,
+        auth=auth,
+        timeout=timeout,
+        allow_redirects=allow_redirects,
+        max_redirects=max_redirects,
+        verify=verify,
+        raise_for_httperror=raise_for_httperror,
+    )
+
+
+def get(
+    url: str,
+    params: Optional[httpx._types.QueryParamTypes] = None,
+    headers: Optional[httpx._types.HeaderTypes] = None,
+    cookies: Optional[httpx._types.CookieTypes] = None,
+    auth: Optional[httpx._types.AuthTypes] = None,
+    allow_redirects: bool = True,
+    max_redirects: Union[_NotSetClass, int] = NOTSET,
+    verify: Union[_NotSetClass, httpx._types.VerifyTypes] = NOTSET,
+    timeout: httpx._types.TimeoutTypes = None,
+    raise_for_httperror: bool = False,
+) -> httpx.Response:
+    """Similar to httpx.get, see the request method for the details.
+
+    allow_redirects is by default True (httpx default value is False).
+    """
+    # pylint: disable=too-many-arguments
+    return request(
+        "GET",
+        url,
+        params=params,
+        headers=headers,
+        cookies=cookies,
+        auth=auth,
+        allow_redirects=allow_redirects,
+        max_redirects=max_redirects,
+        verify=verify,
+        timeout=timeout,
+        raise_for_httperror=raise_for_httperror,
+    )
+
+
+def options(
+    url: str,
+    params: Optional[httpx._types.QueryParamTypes] = None,
+    headers: Optional[httpx._types.HeaderTypes] = None,
+    cookies: Optional[httpx._types.CookieTypes] = None,
+    auth: Optional[httpx._types.AuthTypes] = None,
+    allow_redirects: bool = False,
```

Comment:
This has been set to `True` by default previously (not sure why though).

Reply:
The default value is different between httpx and requests.
* `True` is the [default value for requests](https://github.com/psf/requests/blob/ee93fac6b2f715151f1aa9a1a06ddba9f7dcc59a/src/requests/api.py#L36).
* `False` is the [default value for httpx](https://github.com/encode/httpx/blob/3ba5fe0d7ac70222590e759c31442b1cab263791/httpx/_api.py#L213
)

Since we use httpx, it is easier to follow httpx default values.
SearXNG uses only / mainly (?) two HTTP methods: GET and POST, so there is no impact in changing the default value for OPTIONS.

Note: the parameter name is `allow_redirects` and it's a legacy from the requests. httpx uses `follow_redirects`. I didn't had the courage to rename the parameter in this PR.

Reply:
I see, that makes sense. Thanks for the explanation!

Not really sure why we had even set a different default for the `OPTIONS` request although we don't even use it ...